sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.institutions.controller.getStudentApplicationSummary.e2e-spec.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  createFakeInstitutionLocation,
+  createFakeApplication,
+  saveFakeStudent,
+  createFakeStudentAssessment,
+  createFakeEducationProgramOffering,
+} from "@sims/test-utils";
+import {
+  Application,
+  ApplicationStatus,
+  EducationProgramOffering,
+  Institution,
+  InstitutionLocation,
+  OfferingIntensity,
+  StudentAssessment,
+} from "@sims/sims-db";
+
+describe("StudentInstitutionsController(e2e)-getStudentApplicationSummary", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let collegeC: Institution;
+  let collegeCLocation: InstitutionLocation;
+  let applicationRepo: Repository<Application>;
+  let studentAssessmentRepo: Repository<StudentAssessment>;
+  let institutionLocationRepo: Repository<InstitutionLocation>;
+  let educationProgramOfferingRepo: Repository<EducationProgramOffering>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    // College C.
+    const { institution } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeC = institution;
+    collegeCLocation = createFakeInstitutionLocation(collegeC);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+    applicationRepo = appDataSource.getRepository(Application);
+    studentAssessmentRepo = dataSource.getRepository(StudentAssessment);
+    educationProgramOfferingRepo = dataSource.getRepository(
+      EducationProgramOffering,
+    );
+    institutionLocationRepo = dataSource.getRepository(InstitutionLocation);
+  });
+
+  it("Should get the student application details as summary when student has a submitted application for the institution (application with location id saved).", async () => {
+    // Arrange
+
+    // Student has a submitted application to the institution.
+    const student = await saveFakeStudent(appDataSource);
+    const application = createFakeApplication({
+      location: collegeCLocation,
+      student,
+    });
+    const savedApplication = await applicationRepo.save(application);
+    // Original assessment.
+    const fakeOriginalAssessment = createFakeStudentAssessment({
+      auditUser: student.user,
+    });
+    fakeOriginalAssessment.application = savedApplication;
+    // Offering.
+    const fakeOffering = createFakeEducationProgramOffering({
+      auditUser: student.user,
+    });
+    fakeOffering.offeringIntensity = OfferingIntensity.fullTime;
+    const savedOffering = await educationProgramOfferingRepo.save(fakeOffering);
+    fakeOriginalAssessment.offering = savedOffering;
+    const savedOriginalAssessment = await studentAssessmentRepo.save(
+      fakeOriginalAssessment,
+    );
+    savedApplication.currentAssessment = savedOriginalAssessment;
+    await applicationRepo.save(savedApplication);
```

Comment:
Do we really need this(line 75-90). Is an application for a student with provided location will be sufficient right.?

Given that `createFakeApplication` creates application in submitted status.

Reply:
Technically we don't check the offering details of an application to find if it is linked to a student, so my suggestion was to keep the application data more simple and just adequate for the test case.